integration/networking/port_mapping_linux_test.go

```diff
@@ -408,6 +409,92 @@ func TestAccessPublishedPortFromRemoteHost(t *testing.T) {
 	}
 }
 
+func TestAccessPublishedPortFromCtr(t *testing.T) {
+	skip.If(t, testEnv.IsRootless)
+
+	testcases := []struct {
+		name            string
+		daemonOpts      []string
+		disableBrNfCall bool
+	}{
+		{
+			name: "with-proxy",
+		},
+		{
+			name:       "no-proxy",
+			daemonOpts: []string{"--userland-proxy=false"},
+		},
+		{
+			// Before starting the daemon, disable bridge-nf-call-iptables. It should
+			// be enabled by the daemon because, without docker-proxy, it's needed to
+			// DNAT packets crossing the bridge between containers.
+			// Regression test for https://github.com/moby/moby/issues/48664
+			name:            "no-proxy no-brNfCall",
+			daemonOpts:      []string{"--userland-proxy=false"},
+			disableBrNfCall: true,
+		},
+	}
+
+	// Find an address on the test host.
+	hostAddr := func() string {
+		conn, err := net.Dial("tcp4", "hub.docker.com:80")
```

Comment:
if this needs to be an actual domain, we should consider using one of the designated ones (could be "example.com" here)

Reply:
"example.com" is for documentation, I'm not sure using it in a test that expects it to be actually resolvable and listening is really in the spirit of the thing.

An alternative would be to look up the host's addresses using netlink, and pick one. Which is do-able as this is a Linux-only test. Other tests use this Dial trick now ... if they need to change, I could follow-up with a separate PR?